client/src/components/CheckBox.js

```diff
@@ -26,7 +26,7 @@ export class CheckBox extends React.Component {
             <input
               type="checkbox"
               checked={active}
-              onChange={onClick}
+              onChange={() => !disabled && onClick && onClick(id)}
```

Comment:
> Now that we have a nice syntax (using arrow functions) for bound methods, we should start respecting React best practices and use bound methods on the component for event handlers, instead of in-line methods.

> It's a performance thing. Every time render is called a new method instance is created for this event handler. JS-engine wise that's trivial, but the problem is that functions are compared by memory location. That means, even if nothing has effectively changed about the component with the event handler, React only knows that the event handler methods appear to have changed and that it thinks it needs to update the actual DOM (relative to the JS-engine, this can be a non-trivial performance hit). With a bound method, there's only one instance (and it, of course, compares as equal to itself), so React can properly do its virtual DOM magic.

Reply:
Actually, I think it might be okay to leave it like this since the re-renders will be triggered by props which the function relies on anyways.

Reply:
you're right, I always forget this. Thanks!

Reply:
I didn't know you can pass additional arguments to bind, I thought it was just for `this`! 

I still find this confusing, though. The intent isn't clear.

- binding `this` is the primary use of `.bind`. This function doesn't need that, as arrow-functions are automatically bound. 
- each render call will create a brand new function anyway, so there's no performance gain 

Since this onchange is just passing props onto the handler, couldn't we access those props directly in the handler?

```js
export class CheckBox extends React.Component {
  handleOnClick = () => {
    const { onClick, disabled, id} = this.props
    if (!disabled && onClick) {
      onClick(id);
    }
  };

<input 
  onChange={this.handleOnClick}
/>
```

Reply:
Hmm... I did a little more reading and it apparently `.bind()` actually [creates a new function anyways](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind)

I think Alex is right in that it'd be best to access props inside the handler function